pkg/builder/openapi.go

```diff
@@ -154,6 +155,11 @@ func (o *openAPI) buildDefinitionRecursively(name string) error {
 				schema.Extensions[k] = v
 			}
 		}
+		if v, ok := item.Schema.Extensions[extensionV2Spec]; ok {
```

Comment:
wondering, can we mirror `OpenAPIV2Definition()` output there instead? so we will assume that `OpenAPIDefinition()` will output v3 schemas (which we down-convert by default), with `OpenAPIV2Definition` for cases where the down-conversion is not good enough.

Reply:
or we go the opposite direction: `OpenAPIDefinition` will stay v2, and `OpenAPIV3Definition` can give an advanced v3 schema. That sounds more compatible. We won't have many cases where we need this.

Reply:
the latter sounds better and i took a shot. updated the integration test a bit cover v2 spec generation.

Reply:
the next step is to start v3 builder and verify the v2-extension by it's integration test

Reply:
@yue9944882 I started working on builderv3, I should issue a WIP PR by the end of the week.